pkg/apis/networking/validation/validation.go

```diff
@@ -59,20 +61,39 @@ func ValidateNetworkPolicyName(name string, prefix bool) []string {
 
 // ValidateNetworkPolicyPort validates a NetworkPolicyPort
 func ValidateNetworkPolicyPort(port *networking.NetworkPolicyPort, portPath *field.Path) field.ErrorList {
+
+	allowEndPort := utilfeature.DefaultFeatureGate.Enabled(features.NetworkPolicyEndPort)
```

Comment:
We typically don't have to check feature gates in the validation code. The field should be dropped in the REST strategy code before validation is even called.

Reply:
See Service REST strategy as an example.

Reply:
Hum, OK. I've actually followed the same strategy from @robscott in [here](https://github.com/kubernetes/kubernetes/pull/88503/files#diff-c713e8919642d873fdf48fe8fb6d43e5cb2f53fd601066ff53580ea655948f0dR4223) but will take a look into the REST strategy. If not found will ping you on monday :)

Reply:
Found here, don't know if this is actually the right way (I saw in the services strategy functions to validate if svc==nil, etc) but I'm doing a commit with the changes right now :)

Will try to write some unit tests during this weekend to validate things.

Reply:
I do think the REST strategy is the appropriate solution here even though sometimes validation has been used. I think for AppProtocol I copied the logic from validation somewhere else, but I've since realized that the REST strategy is the more standard place to implement this.

Reply:
Great, thanks Rob. Corrected here